main.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"crypto/sha256"
+	"database/sql"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	_ "github.com/mattn/go-sqlite3"
+)
+
+var db *sql.DB
+
+// Item struct
+type Item struct {
+	ID        int    `json:"id"`
+	Name      string `json:"name"`
+	Category  string `json:"category"`
+	ImageName string `json:"image_name"`
+}
+
+// Initializes Database
+func initDB() {
+	var err error
+	db, err = sql.Open("sqlite3", "./db/mercari.sqlite3")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Create tables if not exist
+	query := `
+	CREATE TABLE IF NOT EXISTS categories (
+		id INTEGER PRIMARY KEY AUTOINCREMENT,
+		name TEXT UNIQUE NOT NULL
+	);
+	CREATE TABLE IF NOT EXISTS items (
+		id INTEGER PRIMARY KEY AUTOINCREMENT,
+		name TEXT NOT NULL,
+		category_id INTEGER NOT NULL,
+		image_name TEXT NOT NULL,
+		FOREIGN KEY (category_id) REFERENCES categories(id)
+	);`
+	_, err = db.Exec(query)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if _, err := os.Stat("uploads"); os.IsNotExist(err) {
+		os.Mkdir("uploads", os.ModePerm)
+	}
+
+	fmt.Println("Database initialized!")
+}
+
+func saveImage(file io.Reader) (string, error) {
+	data, err := io.ReadAll(file)
+	if err != nil {
+		return "", err
+	}
+
+	hash := sha256.Sum256(data)
+	hashString := hex.EncodeToString(hash[:])
+
+	imagePath := filepath.Join("uploads", hashString+".jpg")
+	err = os.WriteFile(imagePath, data, 0644)
+	if err != nil {
+		return "", err
+	}
+
+	return hashString + ".jpg", nil
+}
+
+func getItemsHandler(w http.ResponseWriter, r *http.Request) {
+	rows, err := db.Query(`
```

Comment:
You could also use prepared statements because they tend to have better performance.
https://go.dev/doc/database/prepared-statements